### PR TITLE
add the answer text to the export file

### PR DESCRIPTION
**Pull request by antopalidi**
fixes stadtluzern/decidim-pdf-consultations/issues/40

Screenshot 2023-06-05 at 11 29 29

### CODE REVIEW

File: app/serializers/decidim/participatory_documents/suggestion_serializer.rb [hunk 1]
```diff
@@ -47,6 +48,10 @@ def submitted_on(suggestion)
       def suggestion_body(suggestion)
         truncate(translated_attribute(suggestion.body), length: 50)
       end
+
+      def answer_text(suggestion)
+        truncate(translated_attribute(suggestion.answer), length: 50)
```
Comment (microstudi) [inline app/serializers/decidim/participatory_documents/suggestion_serializer.rb:53]:
```suggestion
        truncate(translated_attribute(suggestion.answer), length: Decidim::ParticipatoryDocuments.max_export_text_length)
```
I'd like to have this `50` number configurable. And, also if it is `zero`, then don't truncate at all the export.

**Review by microstudi — CHANGES_REQUESTED** [review]
Thanks @antopalidi 
this is good, however we can add a configuration for the module to specify the truncate size. See the comment.
Also, make it possible (by setting it to zero) to not truncate at all the export.

File: app/serializers/decidim/participatory_documents/suggestion_serializer.rb (hunk 1 above)
Comment (antopalidi) [reply app/serializers/decidim/participatory_documents/suggestion_serializer.rb]: Thank you for review. Done

File: app/serializers/decidim/participatory_documents/suggestion_serializer.rb [hunk 2]
```diff
@@ -47,6 +48,16 @@ def submitted_on(suggestion)
...
       end
+
+      def answer_text(suggestion)
+        text_length = Decidim::ParticipatoryDocuments.max_export_text_length
+
+        if text_length.positive?
+          truncate(translated_attribute(suggestion.answer), length: text_length)
+        else
+          translated_attribute(suggestion.answer)
+        end
```
Comment (microstudi) [inline app/serializers/decidim/participatory_documents/suggestion_serializer.rb:59]:
```suggestion
        truncated_text(suggestion.answer)
      end
      
      def truncated_text(text)
        text_length = Decidim::ParticipatoryDocuments.max_export_text_length
        return text unless text_length.positive?
        
        truncate(translated_attribute(suggestion.answer), length: text_length)
        end
```

I'd apply this truncation in a more generic way and also apply it to the suggestion_body as well

File: lib/decidim/participatory_documents.rb [hunk 3]
```diff
@@ -4,6 +4,12 @@ module Decidim
   # This namespace holds the logic of the `decidim-participatory_documents` module.
   module ParticipatoryDocuments
     include ActiveSupport::Configurable
+
+    # Public: The maximum length of the answer_text fields to export.
```
Comment (microstudi) [inline lib/decidim/participatory_documents.rb:8]:
```suggestion
    # Public: The maximum length of any text field (body, answers, etc) to export.
```

**Review by microstudi — CHANGES_REQUESTED** [review]
Please add the optional truncation to the body too. With a test. With this we will be fine.

**Review by microstudi — APPROVED** [review]
All good!

**Pull request merged**